### PR TITLE
Talha Sprint4

### PR DESCRIPTION
**Pull request by talha2021skipq**
Hi, For this sprint the main changes are in the following folders:  
1- reactJSApp>src>>index.js/app.js
2- sprint4>>resources>>backendlambda.py
3- sprint4>>sprint4.py(stack)
Thanks

### CODE REVIEW

File: talha/reactJSApp/src/App.js [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+{
+  const res= axios({method: 'PUT',
+  headers:  {  },
+  url:"https://n0q8c9iur5.execute-api.us-east-2.amazonaws.com/prod/" , 
+  data: urltoput});
+//setPosts(res.data);
+ setLoading(false);
+ console.log(res);
+ setPut(true);
+}
```
Comment (shanawar2021skipq) [inline talha/reactJSApp/src/App.js:73]: Great that you have also added delete and put methods in your UI

File: talha/sprint4/resources/backend_lambda.py [hunk 2]
```diff
@@ -32,6 +36,17 @@ def lambda_handler(events, context):
...
+# datares={"Response" : msg, "httpMethod": events['httpMethod'], "body": events['body'] }
+ #return {'statusCode':200 , 'body':json.dumps(datares)}
+ return {
+        
+        'statusCode' : 200,
+        'headers': {
+          'Access-Control-Allow-Headers': '*',
+          'Access-Control-Allow-Origin': '*',
+          'Access-Control-Allow-Methods': '*'
+          },
```
Comment (shanawar2021skipq) [inline talha/sprint4/resources/backend_lambda.py]: Good method to solve CORS error

File: talha/sprint4/sprint4/sprint4_stack.py [hunk 3]
```diff
@@ -25,13 +26,13 @@
...
 
-class TalhaProjectStack(cdk.Stack):
+class Sprint4Stack(cdk.Stack):
 
     def __init__(self, scope: cdk.Construct, construct_id: str, **kwargs) -> None:
         super().__init__(scope, construct_id, **kwargs)
         lambda_role=self.create_lambda_role()
     # The code that defines your stack goes here
-        fixURLtablename="Beta-infraStack-URLTable1792207E-1E3WEGLZJ0NFU"
+        fixURLtablename="Beta-talhaStack-URLTable1792207E-1M8MKFSJFCIUA"
```
Comment (shanawar2021skipq) [inline talha/sprint4/sprint4/sprint4_stack.py]: Try figure out a way of not hardcoding table names

File: talha/sprint4/sprint4/sprint4_stack.py [hunk 4]
```diff
@@ -155,6 +163,7 @@ def create_db_lambda_role(self):
                             aws_iam.ManagedPolicy.from_aws_managed_policy_name('AmazonDynamoDBFullAccess'),
```
Comment (shanawar2021skipq) [inline talha/sprint4/sprint4/sprint4_stack.py]: Just DynamoDBReadWriteAccess would be enough for the sprint. Good for security purpose. I did the same mistake xD

File: talha/reactJSApp/src/ItemPage.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  const items = [
+    { name: 'Apple', quantity: 3 },
+    { name: 'Banana', quantity: 5 },
+    { name: 'Carrot', quantity: 4 },
+    { name: 'Chocolate', quantity: 1 },
+    { name: 'Marshmallow', quantity: 5 },
+    { name: 'Cinnamon', quantity: 10 },
+    { name: 'Lasagna', quantity: 10 },
+    { name: 'Linguine', quantity: 20 }
+  ];
```
Comment (shanawar2021skipq) [inline talha/reactJSApp/src/ItemPage.js:24]: Are these being used in code?

File: talha/reactJSApp/src/App.js [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+  {
+    const fetchPosts= async() => 
+    {
+      setLoading(true);
+      const res=await axios({method: 'GET',headers: { 'Content-Type': 'application/json' },url:"https://n0q8c9iur5.execute-api.us-east-2.amazonaws.com/prod/"});
+      setPosts(res.data);
+      setLoading(false);
+    }
+    fetchPosts();
+  },[]);
```
Comment (shanawar2021skipq) [inline talha/reactJSApp/src/App.js:31]: Add some comments for better understanding.

**Review by shanawar2021skipq — COMMENTED** [review]
Good modifications in your react UI. 
I didn't see AWS Cognito and AWS Amplify code inside infra stack file. If you are using console mention it in comments. 
Good Luck with coding.

File: talha/reactJSApp/src/ItemPage.js (hunk 5 above)
Comment (talha2021skipq) [reply talha/reactJSApp/src/ItemPage.js:24]: No

**Pull request merged**